tests/unit/test_client.py

```diff
@@ -5,11 +5,31 @@
 import pytest
 
 from entitysdk.client import Client
+from entitysdk.config import settings
 from entitysdk.exception import EntitySDKError
 from entitysdk.mixin import HasAssets
 from entitysdk.models.entity import Entity
 
 
+def test_client_api_url():
+    client = Client(api_url="foo")
+    assert client.api_url == "foo"
+
+    client = Client(api_url=None, environment="staging")
+    assert client.api_url == settings.staging_api_url
+
+    client = Client(api_url=None, environment="production")
+    assert client.api_url == settings.production_api_url
+
+    with pytest.raises(
+        EntitySDKError, match="Either the api_url or environment must be defined, not both."
+    ):
+        Client(api_url="foo", environment="staging")
+
+    with pytest.raises(EntitySDKError, match="Neither api_url nor environment have been defined."):
+        Client()
```

Comment:
We could also test the error when environment isn't valid (neither production nor staging)

Reply:
Yeah, I validate the env now in the constructor.